### DISCUSSION TITLE
batch is deprecated?

### DISCUSSION THREAD

**Discussion by bdharrington7** [vitaly-t/pg-promise#874] [Q&A] [answered]
Hi, I saw the latest update 11.4.3 [deprecates the batch function](https://github.com/vitaly-t/pg-promise/releases/tag/11.4.3), but I can't find an issue indicating why this is now deprecated. From the FAQ, using batch is still [the correct way](https://github.com/vitaly-t/pg-promise/wiki/FAQ#why-use-method-batch-instead-of-promiseall) to handle multiple concurrent database calls.

**Comment by vitaly-t** [comment, marked as answer]
I'm moving this to discussions, because Issues is not for asking questions.

Method `batch` was added to make sure that every query request is settled, in the context of auto-managed connections, and it was years before `async/await` syntax became available, which is what you should be using today instead. And as mentioned [in docs](https://vitaly-t.github.io/pg-promise/Task.html#batch), in rare cases where promises need to be created, you can use [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled).

Not all documentation is up-to-date on this yet, but I will do so soon.

**Comment by vitaly-t** [comment]
@bdharrington7 Did this not answer your question? If it did, then please accept the answer.
